patterns/behavioural/inner-trait.md

```diff
@@ -0,0 +1,105 @@
+# Inner Trait
+
+## Description
+
+It is possible to define a private trait that implements all the
+methods of a public trait and also includes some private functions. This pattern
+can be used to provide additional functionality to the implementation of a
+public trait while keeping the private methods hidden from the public API.
+
+## Example
+
+This example demonstrate how a public trait `Car` can be implemented and include
+extra private methods using a auxiliary private trait `InnerCar`.
+
+```rust,ignore
+// trait that is public and part of the API
+pub trait Car {
+    fn get_speed(&self) -> f64;
```

Comment:
I wouldn't recommend using `get_` since `speed` will do, this looks javaish and make it unnecessary long for not much benefits, rust already have types for this, we don't have to spell it out in function name.